Changing The way Coreclr Test Builds

ramarag opened the pull request:
This PR removes all the individual project.json and replaces them with a single  tests\src\Common\test_dependencies\project.json which the tests can build against

@wtgodbe @dagood @rahku @jkotas PTAL

@dotnet/coreclr-contrib  FYI

rahku commented on tests/runtest.proj:
```diff
@@ -354,7 +352,7 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
        In order to avoid regenerating project.json for all the tests projects...I added a common project for all the tests.
-       This project is at tests\src\common\test_runtime\test_runtime.csproj. This project is adding common dependencies to
+       This project is at tests\src\common\test_dependencies\test_dependencies.csproj. This project is adding common dependencies to
        all the tests like dependency on Microsoft.netcore.runtime.coreclr ..etc. Project.json for this project is dynamically 
```
This comment is now confusing. The target actually only refers to test_runtime directory but the comment refers to test_dependencies dir.

ramarag replied:
Yes you are right, the file referred here has not been used for the above purpose for a long time. will remove the file from the comment

rahku commented:
Can I still build an individual test without building whole test repo first?

ramarag commented:
yes you can, i have not regressed any of the existing scenario (hopefully !!)

ramarag commented on tests/src/GC/Performance/Tests/project.json:
```diff
@@ -1,48 +0,0 @@
-{
-  "dependencies": {
```
@Maoni0 @swgillespie  Do you guys really need a  seperate project.json ?

ramarag commented:
@dotnet/jit-contrib i have removed all the project.json under tests\src\Jit\config and replaced them with the two project.json mentioned in this PR. IS there any use of keeping the seperate project.json for JIT ?

ramarag commented:
@dotnet-bot test  Windows_NT x64 Debug Build and Test

ramarag commented:
@dotnet-bot test Windows_NT x64 Debug Build and Test

ramarag commented:
@dotnet-bot  test Windows_NT x86 legacy_backend Checked Build and Test

wtgodbe commented:
@dotnet-bot test CentOS7.1 pri1 please

wtgodbe commented:
@dotnet-bot test OpenSUSE13.2 pri1 please

wtgodbe commented:
@dotnet-bot test Debian8.4 pri1 please

wtgodbe commented:
@dotnet-bot test RHEL7.2 pri1 please

gkhanna79 commented:
CC @pgavlin @russellhadley @RussKeldorph for JIT changes

Overall, this is goodness. I am fine with this change so long as the PR is green and all P1 tests pass.

pgavlin commented:
What does this buy us? In the past, I have found that there is quite a bit of value in having a more precise list of the dependencies for a given test. Does this improve test build times?

CarolEidt commented:
I would prefer not to merge the jit project.json files. I am currently attempting to add support for reference overloads to the SIMD intrinsics, and it is already quite painful to test this. Eliminating any specialized project.json files will make it even more difficult.

wtgodbe commented:
@dotnet-bot test CentOS7.1 pri1 please

adityamandaleeka commented:
I haven't looked at all the files, but from what I've seen so far, your change leaves a ton of empty ItemGroup elements like this (which used to contain project.json and/or app.config):

```
   <ItemGroup>
    </ItemGroup>
```

ramarag commented:
@pgavlin @CarolEidt I have removed the JIT side changes i made except for empty\project.json which i have moved up to be used by any project that wants to build against mscorlib facade

ramarag commented:
@adityamandaleeka we have many such gunk left in the project files. The main motivation behind removing the two lines was to ensure no one authors a new project and expect to use them. The gunk that remain are harmless

adityamandaleeka commented:
@ramarag I am aware that an empty ItemGroup doesn't have a functional impact, but since you're making a bulk change to over 14000 files, I figure you could modify whatever script you're using to remove the elements that are made redundant by your change.

In general, I don't think the fact that there may be some junk in the project files today is justification for adding more when it's easily avoidable.

gkhanna79 commented:
@ramarag It will be good to make things better with each change we make. In that light, is it possible to cleanup the empty ItemGroups @adityamandaleeka is talking about?

ramarag commented:
removed the extra empty props from the project file

ramarag commented:
@dotnet-bot test Debian8.4 pri1 please
@dotnet-bot test RHEL7.2 pri1 please
